test/cpp/c10d/ProcessGroupNCCLErrorsTest.cpp

```diff
@@ -257,3 +297,33 @@ TEST_F(ProcessGroupNCCLErrorsTest, testNCCLErrorsNonBlocking) {
 
   // Communicators might be aborted here, further operations would fail.
 }
+
+TEST_F(ProcessGroupNCCLErrorsTest, testNCCLErrorsNoHeartbeat) {
+  if (skipTest()) {
+    return;
+  }
+
+  ASSERT_TRUE(setenv(c10d::NCCL_BLOCKING_WAIT, "1", 1) == 0);
+  ASSERT_TRUE(setenv(c10d::TORCH_NCCL_HEARTBEAT_TIMEOUT, "3", 1) == 0);
+  ASSERT_TRUE(setenv(c10d::TORCH_NCCL_ENABLE_MONITORING, "1", 1) == 0);
+  auto options = c10d::ProcessGroupNCCL::Options::create();
+  options->timeout = std::chrono::milliseconds(30000);
```

Comment:
just a nit- does this test take 30 sec to run? ideally make it faster

Reply:
(generally 1 sec is still 'a long time' and if you dont feel safe maybe use 2 or 3 but not 10?

Reply:
this is just watchdog timeout it does not affect the time of execution?

Reply:
Running this test only needs 10s.